node/silkworm/stagedsync/util.cpp

```diff
@@ -44,7 +44,11 @@ std::pair<Bytes, Bytes> convert_to_db_format(const ByteView& key, const ByteView
 }
 
 void truncate_table_from(mdbx::cursor& table, Bytes& starting_key, bool reverse) {
-    if (table.lower_bound(db::to_slice(starting_key), false)) {
+    auto current{table.lower_bound(db::to_slice(starting_key), false)};
+    if (reverse) {
+        current = table.to_previous(false);
```

Comment:
What if the cursor is already at the very first position? Will `to_previous` silently do nothing? Better add a unit case.

Reply:
to_previous should return false and current will have nothing it, thus avoiding the entire loop, i can add some unit tests for it though